### DISCUSSION TITLE
useRecoilcallback

### DISCUSSION THREAD

**Discussion by mingiki** [facebookexperimental/Recoil#971] [Q&A]
hello

It's hard to write a callback.
Please provide an example like below

I want process

session setRecoilstate => session Method execution

```
this.setState(
      {
          session: this.OV.initSession(),
      },
      () => {

          //세션 연결
          this.connectToSession();
      },
  );
```

**Comment by drarmstr** [comment]
It appears like this snippet is from a React Class component vs a React Function component.  Could you phrase your example in terms of a functional React component using hooks?

**Comment by mingiki** [comment]
Yes.
I'm using class function callback

I'm currently refactoring the class function to recoil

What I want to do

I want to create and use a function with the changed state after the recoil state is completely changed.

Is there a good way?

**Comment by drarmstr** [comment]
If you'd like to execute a side-effect after a state change consider React Effects [`useEffect()`](https://reactjs.org/docs/hooks-effect.html) or [Atom Effects](https://recoiljs.org/docs/guides/atom-effects).
